src/Nethermind/Nethermind.BeaconNode.OApi/Controllers/ValidatorAttestationPostController.cs

```diff
@@ -0,0 +1,98 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Nethermind.Core2;
+using Nethermind.Core2.Api;
+using Nethermind.Core2.Containers;
+using Nethermind.Logging.Microsoft;
+
+namespace Nethermind.BeaconNode.OApi.Controllers
+{
+    [ApiController]
+    [Route("/validator/attestation")]
+    public class ValidatorAttestationPostController : ControllerBase
```

Comment:
this seems helpful - I think you could help us to fix the routing for the API - this is why I was discussing with Sly that you could look at the OAPI for Eth2

Reply:
> what is the purpose of extracting POST method to external controller with the same route

Basically single responsibility principle.  Each controller maps to one method on the facade.

In this case, the route is the same, but sometimes they are different, e.g. REST collection has a GET /foo/{id} but a POST /foo, different paths. There might be multiple ways to GET, with different criteria, etc, so maybe combine them together, but what happens on POST is very different.

NewAttestationAsync() and PublishAttestationAsync() are different operations; yes they are related because they both deal with blocks, but I don't think that is strong enough reason to put them in the same controller any more than ValidatorDutiesAsync().

I would either do all separate (as I have done), or else maybe separate them along the split with all /node operations in one controller and all /validator in another.

Note that once you get past the BeaconNodeFacade the two operations go very different directions. NewAttestion => AttestationProducer, whilst PublishAttestation => ForkChoice and NetworkPeering.

(The reverse is why combine them in the same controller simply because they have the same path, when the verbs are different, compared to the paths being different.)